segmentation_labeling_app/transforms/transformations.py

```diff
@@ -0,0 +1,212 @@
+    Args:
+        full_array: The video array in numpy format
+        input_fps: The FPS or Hz of the input video
+        output_fps: The desired output FPS or Hz
+        strategy: The down-sampling startegy to follow
+
+    Returns:
+        new_video: A new video down-sampled to desired FPS with selected
+        strategy
+    """
+    if output_fps > input_fps:
+        raise ValueError('Output FPS cannot be greater than input FPS')
+    sampling_ratio = int(input_fps / output_fps)
+    new_frames = []
+    for i in range(0, len(full_array), sampling_ratio):
+        frames = full_array[i:(i+sampling_ratio)]
+        if strategy == 'random':
+            new_frames.append(frames[random.randint(0, len(frames) - 1)])
+        elif strategy == 'maximum':
+            new_frames.append(np.max(frames, axis=0))
+        elif strategy == 'average':
+            new_frames.append(np.mean(frames))
+    return np.stack(new_frames)
+
+
+def get_transformed_center(coordinate_pair: Tuple[int, int],
+                           box_size: Tuple[int, int],
+                           video_shape: Tuple[int, int]):
+    """
+    Returns a coordinate pair that has been transformed to fit as closely to
+    original position as possible but still with box contained totally in video.
+    This is made to shift the coordinate if the subset box goes out of bound of
+    the video shape.
+    Args:
+        coordinate_pair: the original coordinate location for centering
+        box_size: the size of the box for subset
+        video_shape: the shape of the original video
+
+    Returns:
+
+    """
+    if box_size[0] < video_shape[0] and box_size[1] < video_shape[1]:
+        # if the box size is less than frame size it can only be out of bounds in
+        # direction left and right and one direction up and down
+        transformed_x = coordinate_pair[0]
+        transformed_y = coordinate_pair[1]
+        # can't be centered left
+        if coordinate_pair[0] < (math.ceil(box_size[0]/2) - 1):
+            transformed_x = math.ceil(box_size[0] / 2) - 1
+
+        # can't be centered up
+        if coordinate_pair[1] < (math.ceil(box_size[1]/2) - 1):
+            transformed_y = math.ceil(box_size[1] / 2) - 1
+
+        # can't be centered right
+        if (coordinate_pair[0] + (math.ceil(box_size[0]/2))) > video_shape[0] - 1:
+            transformed_x = video_shape[0] - math.ceil(box_size[0] / 2) - 1
+
+        # can't be centered down
+        if (coordinate_pair[1] + (math.ceil(box_size[1]/2))) > video_shape[1] - 1:
+            transformed_y = video_shape[1] - math.ceil(box_size[1] / 2) - 1
+
+        return transformed_x, transformed_y
+
+    else:
+        raise ValueError('Box is larger or equal in size in one or more'
+                         'dimensions to original video shape. Provide a '
+                         'box size smaller than original video.')
+
+
+def get_centered_coordinate_box(coordinate_pair: Tuple[int, int],
```

Comment:
I may ask you to rework this a bit if the video transformations take too long, but performance isn't our number one concern since this isn't going to be used very often.